BUILD_LOG_MACROS_USED

peterbe opened the pull request:
Friday fun(?)
If you set `export BUILD_LOG_MACROS_USED=/tmp/macros-used.log` and then run a complete `yarn build` with the ~11k en-us docs and then write [a python script](https://gist.github.com/peterbe/ee3e14ceb202ad08ca4425e82d89976f)
You get this juicy output:

<details>
MOST POPULAR...
domxref                        35,351
specname                       10,093
cssxref                        9,490
spec2                          8,755
compat                         8,286
jsxref                         7,597
apiref                         5,329
htmlelement                    4,500
glossary                       3,067
bug                            3,019
embedlivesample                2,532
svgattr                        2,260
svgelement                     2,171
event                          1,829
webextapiref                   1,479
seecompattable                 1,392
htmlattrxref                   1,145
cssref                         981
embedinteractiveexample        972
httpheader                     954
anch                           836
optional_inline                689
htmlattrdef                    665
addonsidebar                   664
interface                      655
jsref                          646
csssyntax                      623
interwiki                      610
non-standard_header            516
webextexamples                 511
cssinfo                        505
readonlyinline                 504
embedghlivesample              493
page                           447
securecontext_header           402
draft                          360
deprecated_header              350
svgref                         328
rfc                            322
obsolete_header                302
httpsidebar                    287
learnsidebar                   285
availableinworkers             265
jssidebar                      259
inheritancediagram             251
httpstatus                     227
deprecated_inline              197
gecko_minversion_inline        194
userlink                       193
templatelink                   192
defaultapisidebar              187
mathmlelement                  184
experimental_inline            181
sectiononpage                  180
ariarole                       177
htmlref                        173
previousmenunext               162
obsolete_inline                160
mdnsidebar                     159
toolssidebar                   156
httpmethod                     144
firefoxsidebar                 142
non-standard_inline            134
embedyoutube                   134
csp                            132
quicklinkswithsubpages         123
includesubnav                  116
xsltref                        105
previousnext                   92
svginfo                        85
geckorelease                   81
firefox_for_developers         76
gamessidebar                   73
tag                            72
no_tag_omission                72
ifmethod                       69
js_property_attributes         49
source                         45
gecko                          43
note                           36
mathmlref                      36
htmlsidebar                    36
gecko_minversion_header        36
jsfiddleembed                  35
exception                      29
securecontext_inline           27
nextmenu                       26
webrtcsidebar                  25
fx_minversion_inline           25
fx_minversion_header           25
previousmenu                   23
compatgeckodesktop             23
landingpagelistsubpages        22
index                          21
css_key_concepts               21
xulelem                        20
webglsidebar                   19
unimplemented_inline           18
htmlversioninline              18
canvassidebar                  18
spec                           17
listsubpages                   17
htmlreftable                   15
subpageswithsummaries          14
listsubpagesforsidebar         14
xulattr                        13
ifattribute                    13
domattributemethods            13
webassemblysidebar             12
livesamplelink                 12
mediawiki.external             11
embednewrelicchart             11
embedtest262reportresultstable 10
interfaceoverview              9
warning                        8
pref                           8
editorguidequicklinks          8
compatchrome                   7
todo                           6
next                           6
webkitbug                      5
previous                       5
outdated                       5
glossarydisambiguation         5
fx_minversion_note             5
discussionlist                 5
xref_cssvisual                 4
manch                          4
link                           4
learnbox                       4
glossarylist                   4
wiki.localize                  3
tenthcampaignquote             3
xref_cssinitial                2
xref_cssinherited              2
xref_csscomputed               2
serviceworkersidebar           2
ref                            2
listgroups                     2
h3_gecko_minversion            2
compatopera                    2
compatno                       2
webextallexamples              1
webextallcompattables          1
property_prefix                1
obsoletebadge                  1
nonstandardbadge               1
docstatusquicklinks            1
cssanimatedproperties          1
css_ref                        1
compatversionunknown           1
compatunknown                  1
compatoperamobile              1
compatgeckomobile              1
apilistalpha                   1


DELETE WORTHY (never used)
kumascript/macros/訳語.ejs
kumascript/macros/訳注.ejs
kumascript/macros/英語版章題.ejs
kumascript/macros/原語併記.ejs
kumascript/macros/xref2.ejs
kumascript/macros/xref.ejs
kumascript/macros/WebExtAPISidebar.ejs
kumascript/macros/web.link.ejs
kumascript/macros/unimplementedGeneric.ejs
kumascript/macros/tree.ejs
kumascript/macros/SpiderMonkeySidebar.ejs
kumascript/macros/SimpleBanner.ejs
kumascript/macros/SimpleBadge.ejs
kumascript/macros/secureContextGeneric.ejs
kumascript/macros/propertiesbox.ejs
kumascript/macros/previousPage.ejs
kumascript/macros/Preferences_System_TOC.ejs
kumascript/macros/optionalGeneric.ejs
kumascript/macros/ObsoleteGeneric.ejs
kumascript/macros/NoteStart.ejs
kumascript/macros/NoteEnd.ejs
kumascript/macros/Non-standardGeneric.ejs
kumascript/macros/nextPage.ejs
kumascript/macros/minversionGeneric.ejs
kumascript/macros/MakeColumnsForDL.ejs
kumascript/macros/LXRSearch.ejs
kumascript/macros/LocalizationStatusInSection.ejs
kumascript/macros/LiveSampleURL.ejs
kumascript/macros/LegacyAddonsNotice.ejs
kumascript/macros/languages.ejs
kumascript/macros/jsOverrides.ejs
kumascript/macros/jsctypesSidebar.ejs
kumascript/macros/jsapixref.ejs
kumascript/macros/jsapi_minversion_inline.ejs
kumascript/macros/jsapi_minversion_header.ejs
kumascript/macros/jsapi-requires-request.ejs
kumascript/macros/IRCLink.ejs
kumascript/macros/HTML5ArticleTOC.ejs
kumascript/macros/h2_gecko_minversion.ejs
kumascript/macros/h1_gecko_minversion.ejs
kumascript/macros/gecko_minversion_note.ejs
kumascript/macros/gecko_callout_heading.ejs
kumascript/macros/FontAwesomeIcon.ejs
kumascript/macros/ExperimentalBadge.ejs
kumascript/macros/endnote.ejs
kumascript/macros/DOM0.ejs
kumascript/macros/DeprecatedGeneric.ejs
kumascript/macros/DeprecatedBadge.ejs
kumascript/macros/CSSTutorialTOC.ejs
kumascript/macros/CompatSafari.ejs
kumascript/macros/CompatIE.ejs
kumascript/macros/CompatibilityTable.ejs
kumascript/macros/CompatAndroid.ejs
kumascript/macros/B2GOnlyHeader2.ejs
</details>

If you count that carefully you see it's **54 kumascript macros** that we can now delete. Wow!

peterbe commented:
@escattone It's not a lot of pain to have those 54 macros laying around but I still think we should delete them. Because if we ever need them back, that's what git's elephant-memory is all about. 

But I made the above calculation based on the old content that was *inside* `yari`. That's gone now and we have a fresh dump in Content which is from Aug 18 2020. 

But perhaps we just wait even further down into the year or even after Yari1.

peterbe commented:
@escattone We have a pretty fresh copy of the prod DB (Aug 30th I think). Next week, I can do another refresh of mdn/content and then we can use that to justify deleting about ~50 macros that are never ever used.

escattone commented on kumascript/src/render.js:
```diff
@@ -210,6 +211,13 @@ async function render(
       if (!priorResult) {
         signatureToResult.set(token.signature, currentResult);
       }
+      if (process.env.BUILD_LOG_MACROS_USED) {
+        fs.appendFileSync(
+          process.env.BUILD_LOG_MACROS_USED,
+          `${macroName}|${token.args}\n`,
+          "utf-8"
+        );
+      }
```
This won't work. It doesn't account for macros called only from within other macros. You want to move this code right after the following line in `kumascript/src/templates.js`:

https://github.com/mdn/kumascript/blob/affb2a7ca9f9789006111e97c71a8d57cf7f5762/src/templates.js#L94

escattone reviewed (changes requested):
@peterbe Sorry for the delay in reviewing this! This is a great idea, but the approach is flawed. It doesn't take into account the macros that are never used directly within documents but only used within other macros, which is the case for many of the macros that you've identified as never used. See below for the suggested changes.

peterbe commented:
I also renamed the env var because otherwise it looks like you're supposed to set the value to `true` or something. 

Unfortunately, I can't give an update on the macros used. But, I've updated https://gist.github.com/peterbe/ee3e14ceb202ad08ca4425e82d89976f and I'll run it once we get over https://github.com/mdn/yari/issues/1338

escattone commented on docs/envvars.md:
```diff
@@ -128,12 +128,12 @@ When doing local development, it's recommended to set this to
 The base URL used in the Interactive Example iframes.
 
-### `BUILD_LOG_MACROS_USED`
+### `BUILD_LOG_MACROS_USED_LOGFILE`
```
@peterbe I was thinking that `BUILD_MACROS_USED_LOGFILE` might be less verbose and better? What do you think?

peterbe replied:
Updated. r?

escattone reviewed (approved):
@peterbe Looks and works great now, thanks! Just a suggestion for the env var name, but up to you.
